pkg/sql/conn_executor_exec.go

```diff
@@ -775,9 +775,11 @@ func (ex *connExecutor) handleAOST(ctx context.Context, stmt tree.Statement) err
 			}
 			return errors.AssertionFailedf("expected bounded_staleness set with a max_timestamp_bound")
 		}
-		return errors.AssertionFailedf(
+		return pgerror.Newf(
+			pgcode.FeatureNotSupported,
```

Comment:
could also be InvalidParameter, but up to you

Reply:
i'm keeping it as this to be consistent with the other places